### PR TITLE
Auto-detect TARGET_OS for mac in make-adopt-build-farm.sh

### PR DESCRIPTION
**Pull request by sxa**
mac was not being auto-detected correctly. Also wrong variable was used for the solaris detection, so I'm fixing that here too.

Signed-off-by: Stewart X Addison <sxa@redhat.com>

### CODE REVIEW

File: build-farm/make-adopt-build-farm.sh [hunk 1]
```diff
@@ -39,8 +39,9 @@ fi
 ## so needs to be special cased - on everthing else "uname" is valid
 if [ -z "$TARGET_OS" ]; then
   TARGET_OS=$(uname)
-  if [ "$OSTYPE" = "cygwin" ]; then TARGET_OS=windows ; fi
-  if [ "$OSTYPE" = "SunOS"  ]; then TARGET_OS=solaris ; fi
+  if [ "$OSTYPE"    = "cygwin" ]; then TARGET_OS=windows ; fi
+  if [ "$TARGET_OS" = "SunOS"  ]; then TARGET_OS=solaris ; fi
```
Comment (karianna) [inline build-farm/make-adopt-build-farm.sh:43]: This SunOs logic changed.  Was that deliberate?

**Review by karianna — CHANGES_REQUESTED** [review]

File: build-farm/make-adopt-build-farm.sh (hunk 1 above)
Comment (sxa) [reply build-farm/make-adopt-build-farm.sh:43]: Yeah I considered doing that in a separate PR but (mostly becuase hardly anyone would care about Solaris) I put it in here. Happy to split it out again and get a free bonus PR out of it.

`$OSTYPE` won't be set to `SunOS` by default on Solaris so this needs to be using the output from `uname` - stored in `$TARGET_OS` - defined a few lines above to have the desired effect.

Comment (sxa) [reply build-farm/make-adopt-build-farm.sh:43]: (Also note I did mention that I'd changed solaris in the PR description)

**Review by karianna — APPROVED** [review]

**Comment by sxa** [comment]
Extra commit added with the slightly odd looking `ARCHITECTURE` conversion from `arm` to `aarch64` for macos. It looks quite odd since the next line converts `armv7l` to `arm` for 32-bit Linux/arm ...

**Comment by sxa** [comment]
Fixes https://github.com/adoptium/temurin-build/issues/2849

**Comment by sxa** [comment]
@andrew-m-leonard can you approve this one too (You're probably in a good position to check it)

FYI I noticed in https://github.com/adoptium/temurin-build/wiki/Building-OpenJDK-using-temurin-build-scripts-within-the-adopt-build-docker-container you mention setting a few variables like OS and architecture explicitly. In general those are now autodetected so the process is simpler now (Except on macos until this pr is merged ;-) )

**Review by andrew-m-leonard — APPROVED** [review]
lgtm

**Comment by sxa** [comment]
check failures are unrelated and have issues elsewhere to deal with them, therefore merging

**Pull request merged**